sql/src/main/java/io/crate/planner/consumer/QueryAndFetchConsumer.java

```diff
@@ -198,11 +198,13 @@ private static boolean tryApplyOrderAndLimitPreFetch(QuerySpec qs,
                                                          Limits limits) {
         OrderBy orderBy = qs.orderBy().orElse(null);
         if (orderBy == null) {
-            TopNProjection topN = new TopNProjection(
-                limits.finalLimit(),
-                limits.offset(),
-                InputColumn.fromSymbols(fetchDescription.preFetchOutputs()));
-            plan.addProjection(topN, null, null, null);
+            if (limits.hasLimit()) {
```

Comment:
this is not really relevant for this PR - I noticed this when I tried to remove the joins on subQuery restriction - should I revert it for now?